src/instrumentation.rs

```diff
@@ -0,0 +1,98 @@
+use crate::ffi;
+use crate::pyclass::boolean_struct::False;
+use crate::types::PyFrame;
+use crate::{Bound, PyAny, PyClass, PyObject, PyRefMut, PyResult, Python};
+use std::ffi::c_int;
+
+pub trait Event<'py>: Sized {
```

Comment:
What is the trait useful for?

Reply:
That was part of the implementation suggested in https://github.com/PyO3/pyo3/issues/4008#issuecomment-2027313230. I think you're right that it's not needed with the approach I'm aiming for though.